### PR TITLE
Add interpolation to MinimalTsit5 and cleanify the code more

### PR DESCRIPTION
**Pull request by Datseris**
To-dos

- [x] Change all broadcasts to `for` loops, as instructed by @YingboMa 
- [x] Add interpolation (valid between `tprev` and `t`)
- [x] Hook it up to `DiffEqBase`
- [x] Think about what to do with the adaptive step, by comparing with existing DiffEq code using different norms
- [x] Add tests
- [x] Allow modifying current state
- [ ] Do proper benchmarks

*I will be copying this code to `DynamicalSystemsBase` as well, as I intend to move the `DiffEq` dependency to `DynamicalSystems` behind `@require`*

Then, I'll also add step methods for `Vector{Vector}` and `Vector{SVector}`.

Edit: here is how adaptive is _currently_ done: 
https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/master/src/integrators/integrator_utils.jl#L156-L179
https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/master/src/integrators/integrator_utils.jl#L292-L350
https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/master/src/integrators/integrator_utils.jl#L7-L25
(it's not going to be a simple copy-paste)

### CODE REVIEW

File: src/simple_tsit5_integrator/benchmarks.jl [hunk 1]
```diff
@@ -1,4 +1,23 @@
 using BenchmarkTools, OrdinaryDiffEq
+
+function loop(u, p, t)
```
Comment (ChrisRackauckas) [inline src/simple_tsit5_integrator/benchmarks.jl]: there's a test folder :)

File: src/simple_tsit5_integrator/tsit5_integrator.jl [hunk 2]
```diff
@@ -38,45 +44,62 @@ function _build_caches(::MinimalTsit5, ::Type{T}) where {T}
...
+
+    return cs, as, rs
 end
 
+#######################################################################################
+# Stepping
+#######################################################################################
 # IIP version for vectors and matrices
-function DiffEqBase.step!(integ::MinimalTsit5Integrator{true, T, S}) where {T, S}
+function step!(integ::MinimalTsit5Integrator{true, T, S}) where {T, S}
```
Comment (ChrisRackauckas) [inline src/simple_tsit5_integrator/tsit5_integrator.jl]: this should extend

Comment (Datseris) [reply src/simple_tsit5_integrator/tsit5_integrator.jl]: I do `import DiffEqBase: init, step!, isinplace`

Comment (ChrisRackauckas) [reply src/simple_tsit5_integrator/tsit5_integrator.jl]: yeah I removed that because it's against our style guide to extend without reference.

**Pull request merged**